bigtable_rs/src/bigtable/read_rows.rs

```diff
@@ -33,6 +35,35 @@ pub async fn decode_read_rows_response(
     Ok(rows)
 }
 
+/// Decode the stream of `ReadRowsResponse` into a stream of `Result<(RowKey, Vec<RowCell>)>>`.
+pub async fn decode_read_rows_response_stream<'a>(
+    timeout: &'a Option<Duration>,
```

Comment:
Are you sure a timeout is really wanted here?
Streams are supposed to live as long as there is data flowing. By adding a timeout we risk ending the stream sooner than expected.

Reply:
I guess this is for users to specify a guardrail on how long they would expect a row read call to last. The same timeout is set on the [endpoint configuration](https://docs.rs/tonic/latest/tonic/transport/channel/struct.Endpoint.html#method.timeout) via [here](https://github.com/liufuyang/bigtable_rs/blob/c88bc67d50d0503d74e4c7444eb994cdf5a3a2ff/bigtable_rs/src/bigtable.rs#L307)

So perhaps that is fine? For users want forever timeout they can start the client with timeout as None?

Reply:
After a second thought, perhaps you are correct that when the stream API is used, the timeout option should be out of the way. This could help in the case if the client is setup with a timeout, and non-stream API can respect that, while the stream-APIs just ignores that one.

Reply:
And if we remove this timeout parameter here, this method perhaps can be simplified as:
```rust
/// Decode the stream of `ReadRowsResponse` into a stream of `Result<Vec<Result<(RowKey, Vec<RowCell>)>>>>`.
pub async fn decode_read_rows_response_stream<'a>(
    rrr: Streaming<ReadRowsResponse>,
) -> Pin<Box<dyn Stream<Item = Result<Vec<Result<(RowKey, Vec<RowCell>)>>>> + Send + 'a>> {
    let output = rrr.map(|result| {
        result
            .map(|read_rows_response| decode_read_rows_response_to_vec(read_rows_response.chunks))
            .map_err(Error::from)
    });
    return Box::pin(output);
}
```

Reply:
Ah yes, this makes sense—in this case the caller can set its own timeout if needed.

Removing timeout cleans things up significantly.